presto-main/src/main/java/com/facebook/presto/sql/planner/NodePartitionMap.java

```diff
@@ -75,10 +72,10 @@ public Node getNode(Split split)
 
     public BucketNodeMap asBucketNodeMap()
     {
-        Map<Integer, Node> bucketToNode = new HashMap<>(bucketToPartition.length);
-        for (int i = 0; i < bucketToPartition.length; i++) {
-            bucketToNode.put(i, partitionToNode.get(bucketToPartition[i]));
+        ImmutableList.Builder<Node> bucketToNode = ImmutableList.builder();
+        for (int partition : bucketToPartition) {
```

Comment:
Alternatively, we could make the iteration over `bucketId` explicit, i.e.
```
for (bucket i = 0; i < bucketToPartition.length) {
    bucketToNode.add(partitionToNode.get(bucketToPartition[bucket]))
}
```

This explicitly says `bucketToNode(bucket) = partitionToNode(bucketToPartition(bucket))`. Although I don't feel strongly here.

Reply:
I considered that, but I'm not sure it improves readability, and IDEA actually warns that it can be replaced with foreach. But it's a good thought.